File: tests/fixtures/sample.kt
package sample.app

import kotlin.collections.List
import java.time.Instant as TimeInstant

interface Greeter {
    fun greet(name: String): String
}

data class User(val id: String, val name: String) : Greeter {
    override fun greet(name: String): String {
        return "Hello, $name"
    }
}

sealed class Result

object Registry {
    fun lookup(id: String): User? = null
}

fun topLevel(count: Int = 0): Unit {}

File: tests/fixtures/sample.scala
package sample.app

import scala.collection.mutable.ListBuffer
import java.time.Instant
import scala.util.{Try, Success, Failure}

trait Greeter {
  def greet(name: String): String
}

case class User(id: String, name: String) extends Greeter {
  override def greet(name: String): String = {
    s"Hello, $name"
  }
}

sealed trait Result
sealed class Error(msg: String)

object Registry {
  def lookup(id: String): Option[User] = None
}

class Service {
  def run(count: Int): Unit = {}
}

def topLevel(x: Int): Int = x + 1
